pandas/core/window/rolling.py

```diff
@@ -374,16 +375,34 @@ def _wrap_result(self, result, block=None, obj=None):
         index = obj.index
 
         if isinstance(result, np.ndarray):
-
             if result.ndim == 1:
                 from pandas import Series
 
                 return Series(result, index, name=obj.name)
+            elif result.ndim == 2 and columns is not None:
```

Comment:
I think this should be handled by `return type(obj)(result, index=index, columns=block.columns)`

Reply:
I think it breaks if this is not there? I would just try changing this and see ... this whole rolling.py API was a scratch. Somebody who knows the intentions should probably just either undo the rolling.py diff and quickly make the changes or unpick the stuff. There were many things where it is was not possible to really know what was intended there. Will push the other changes shortly.